NetSSL_OpenSSL/src/Context.cpp

```diff
@@ -921,18 +957,22 @@ void Context::initDH(bool use2048Bits, const std::string& dhParamsFile)
 
 #else // OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
 
-		if (use2048Bits)
+		if (keyDHGroup == KEY_DH_GROUP_2048)
 		{
 			dh->p = BN_bin2bn(dh2048_p, sizeof(dh2048_p), 0);
 			dh->g = BN_bin2bn(dh2048_g, sizeof(dh2048_g), 0);
 			dh->length = 256;
 		}
-		else
+		else if (keyDHGroup == KEY_DH_GROUP_1024)
 		{
 			dh->p = BN_bin2bn(dh1024_p, sizeof(dh1024_p), 0);
 			dh->g = BN_bin2bn(dh1024_g, sizeof(dh1024_g), 0);
 			dh->length = 160;
 		}
+		{
```

Comment:
This must be in else: https://github.com/pocoproject/poco/commit/c4f66d518891736d08636fa0fa7758303587899c#r152747523

Issue link: https://github.com/pocoproject/poco/issues/4877